File: wed.py
# wed.py
def chkFrom(i, ans):
    fighters = 0
    hash = {}
    while(i < n):
        ele = arr[i]
        try:
            count = hash[ele]
            if(fighters == 0):
                j = i
                j = i
            fighters += 1
            if(count == 1):
                fighters += 1
            hash[ele] += 1
            if(fighters > k):
                ans += k
                return chkFrom(j, ans)
        except:
            hash[ele] = 1
        i += 1
    return ans + fighters

test = int(input())
while(test):
    test -= 1
    n, k = list(map(int, input().split()))
    arr = input().split()
    print(chkFrom(0, k))

    exit(0)
    i = n - 1
    fighters = 0
    ans = k
    hash = {}


    while(i >= 0):
        ele = arr[i]
        try:
            count = hash[ele]
            hash[ele] += 1
            fighters += 1
            if(count == 1):
                fighters += 1

            if (fighters > k):
                ans += k
                fighters = 0
                hash = {}
                hash[ele] = 1
        except:
            hash[ele] = 1
        i -= 1
            
    
    print(ans + fighters)

File: wed.c
#include<stdio.h>

int minimize(int* arr, int *n){
    int i = 0, j, count = 0, temp;
    for (j = 0; j < *n; j++){
        temp = arr[j];
        if(temp > 1){
            arr[i] = temp;
            i++;
            count += temp;
        }
    }
    *n = i;
    return count;
}

void addTable(int* arr, int n){
    for (int i = 0; i < n; i++){
        arr[i]--;
    }
}


int main(){
    int test;
    scanf("%d", &test);
    int n, cost, ans, temp, fighters, i;
    while(test--){
        int hash[1001];
        scanf("%d%d",&n, &cost );
        int arr[n];
        for(i = 0; i < n; i++){
            scanf("%d", &arr[i]);
        }
        ans = cost;
        i = 0;
            hash = {0};
        while(i < n){
            if(hash[arr[i]] != 0){
                cost++;
                hash = {0};
            }
            else{
                i++;
            }

        }



    printf("%d\n", ans);
    }
    return 0;
}